### PR TITLE
Add tools 15 and XCORE AI testing support

### PR DESCRIPTION
**Pull request by ed-xmos**
Add automated testing on XS3/AI sim and HW

### CODE REVIEW

**Review by mbanth — COMMENTED** [review]
What I have seen looks sensible.  However, the Jenkinsfile and the two Makefiles deserve another set of eyes on them.

**Comment by ed-xmos** [comment]
Note the Jenkins agent label "srv-bri-nuc2" needs to be updated as soon as Brennan has setup the new NUCs for xcore-ai agents. Currently it runs on the lib_voice_front_end NUC which will blocked for many hours at a time.

I'd also recommend a squash commit on this because of the number of commits I had to make to work out the groovy syntax for running two agents..

File: Jenkinsfile [hunk 1]
```diff
@@ -1,52 +1,135 @@
...
-    success {
-      updateViewfiles()
-    }
-    cleanup {
-      xcoreCleanSandbox()
-    }
-  }
+        stage('XCORE AI tests') {
+          agent {
+            label "srv-bri-nuc2"
```
Comment (BrennanGit) [inline Jenkinsfile:68]: The labels "xs3" and "xcore.ai-explorer" are now available on all the relevant machines

File: Jenkinsfile [hunk 2]
```diff
@@ -1,52 +1,135 @@
...
+          }
+          environment {
+            // '/XMOS/tools' from get_tools.py and rest from tools installers
+            TOOLS_PATH = "/XMOS/tools/${params.AI_TOOLS_VERSION}/XMOS/xTIMEcomposer/${params.AI_TOOLS_VERSION}"
+            REPO = 'lib_logging'
```
Comment (BrennanGit) [inline Jenkinsfile:73]: You could maybe move the REPO and VIEW vars into a top-level environment block.

**Review by BrennanGit — APPROVED** [review]
Once that label change is in, this looks good. There's maybe a couple of ways to make parallel agents like this a bit cleaner by collecting more at the start but that's really all down to readability which isn't really a problem here.

**Comment by mbanth** [comment]
> Once that label change is in, this looks good. There's maybe a couple of ways to make parallel agents like this a bit cleaner by collecting more at the start but that's really all down to readability which isn't really a problem here.

Since lib_logging is serving as a Golden Reference for publishing the other libraries, it would be good to have the Jenkinsfile as clean as possible.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,52 +1,135 @@
...
+          stages {
+            stage('Checkout') {
+              steps {
+                checkout scm
+              }
+            }
+            stage('Install Dependencies') {
+              steps {
+                sh '/XMOS/get_tools.py ' + params.AI_TOOLS_VERSION
+                installDependencies()
```
Comment (BrennanGit) [inline Jenkinsfile:88]: I would change this to installPipfile since the nucs don't have brew installed and we've removed the need for cpanfiles.

**Comment by BrennanGit** [comment]
I have made some updates in a separate branch/PR and here is the Jenkins results. http://srv-bri-jcim0:8080/job/XMOS/job/lib_logging/job/PR-41/
I'm seeing the odd job go very slowly in unexpected places (eg reading a file from disk) so merge with caution.

**Comment by ed-xmos** [comment]
Superseded by https://github.com/xmos/lib_logging/pull/42

**Pull request closed without merging**